connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -193,5 +247,9 @@ service SparkConnectService {
 
   // Analyzes a query and returns a [[AnalyzeResponse]] containing metadata about the query.
   rpc AnalyzePlan(AnalyzePlanRequest) returns (AnalyzePlanResponse) {}
+
+  // Add artifacts to the session and returns a [[AddArtifactsResponse]] containing metadata about
+  // the added artifacts.
+  rpc AddArtifacts(stream AddArtifactsRequest) returns (AddArtifactsResponse) {}
```

Comment:
There's some somewhat concerning language about stream performance in Python on gRPC's [performance page](https://grpc.io/docs/guides/performance/), can you provide any benchmarks about how well this performs for JVM/Python clients? Individual JAR files, and the number of JAR files needed, can become quite large.

Reply:
AFAIK that is only problem when you stream data to a python process, and not the other way around.

Reply:
Cool. Some benchmarking would still be nice at some point, even on JVM, but that's good to hear.